inc/WebhookHandler/GitHub.php

```diff
@@ -50,7 +50,7 @@ public function permission_callback(): ?bool {
 
 		$secret = $this->get_secret( $project );
 
-		$payload_signature = 'sha1=' . hash_hmac( 'sha1', $this->request->get_body(), $secret );
+		$payload_signature = 'sha1=' . hash_hmac( 'sha1', wp_json_encode( $params ), $secret );
```

Comment:
Why do we have to change this?

Reply:
So we only consume the body params, never the body itself. And we only set the params in tests too, so I assume that’s how it’s supposed to work.

That means we never use the body itself, so get_body() is always null, so we were calculating the hash of nothing.

I spotted it because of some PHP deprecation warnings, as it‘s not allowed to pass null to this function, only strings.

Reply:
It's just odd that the current way is working since the body shouldn't be `null`. I think we actually have to update the tests to set the body like done in https://github.com/WordPress/wordpress-develop/blob/13d5244c43d65da5fd3ff7abd390fa36bb4a87b5/tests/phpunit/tests/rest-api/rest-request.php#L895-L901.

Reply:
Let me try that